medallion/license_service.py

```diff
@@ -0,0 +1,16 @@
+class LicenseService:
+    _mandiant_license = "mandiant"
+    _nozomi_license = "nozomi"
+
+    @classmethod
+    def can_user_read_collection(cls, user: dict | None, collection_license: str) -> bool:
+        if user is None or (user_license := user.get("license")) is None:
+            return False
+
+        if user_license == cls._mandiant_license:
```

Comment:
if we happen to add a third license we will have to change this condition, can we make it more explicit?

such as 

```
switch collection_license
  nozomi:
    return user_license in (nozomi, mandiant)
  mandiant
     return user_license in (mandiant)
  default
      retutn False
```


even better you could create a static mappgin of collection_licenses -> user licences and do something like

```user_license in mappings[collection_licenses]```